index.js

```diff
@@ -179,7 +179,7 @@ Request.prototype.run = function () {
                 }
             }
 
-            if (res.status >= 400) {
+            if (!res.status || res.status < 100 || res.status >= 400) {
```

Comment:
Shouldn't the code also set an appropriate status code in case it's missing or is `< 100` ? In this block we're just re-throwing it.